modules/chimera/src/main/java/org/dcache/chimera/JdbcFs.java

```diff
@@ -725,36 +719,40 @@ public FsInode path2inode(String path, FsInode startFrom) throws ChimeraFsExcept
     @Override
     public String inode2id(FsInode inode) throws ChimeraFsException {
         try {
-            return _idCache.get(inode.ino(), () -> {
+            return _idCache.get(inode.ino(), (key) -> {
                 String id = _sqlDriver.getId(inode);
                 if (id == null) {
-                    throw FileNotFoundChimeraFsException.of(inode);
+                    throw new CompletionException(FileNotFoundChimeraFsException.of(inode));
                 }
                 return id;
             });
-        } catch (ExecutionException e) {
-            Throwables.throwIfInstanceOf(e.getCause(), ChimeraFsException.class);
-            Throwables.throwIfInstanceOf(e.getCause(), DataAccessException.class);
-            Throwables.throwIfUnchecked(e.getCause());
-            throw new RuntimeException(e.getCause());
+        } catch (CompletionException e) {
+            if (e.getCause() != null) {
```

Comment:
I think the check is unnecessary. CompletionException should always have a cause.